src/basics.md

```diff
@@ -470,6 +472,50 @@ fn main() {
 }
 ```
 
+[ex-extract-hashtags]: #ex-extract-hashtags
+<a name="ex-extract-hashtags"></a>
+## Extract a list of unique #Hashtags from a text
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Extracts a sorted and deduplicated list of hashtags from a text.
+
+The hashtag regex given here only catches latin hashtags that start with a letter. The [twitter hashtag regex] is way more complicated.
+
+[twitter hashtag regex]: https://github.com/twitter/twitter-text/blob/master/java/src/com/twitter/Regex.java#L255
+
+```rust
+extern crate regex;
+#[macro_use] extern crate lazy_static;
+
+use regex::Regex;
+
+fn extract_hashtags(text: &str) -> Vec<&str> {
+    lazy_static! {
+        static ref HASHTAG_REGEX : Regex = Regex::new(
+                r"\#[a-zA-Z][0-9a-zA-Z_]*"
+            ).unwrap();
+    }
+    let mut results = vec![];
```

Comment:
why not just map and collect into HashSet or BTreeSet  (if you need sorted results)?